File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
  
      int curMax = 0, globalMax = INT_MIN;
       int n = nums.size(); 
      for(int i = 0 ; i < n ; i++) {
            curMax += nums[i];
            
          if(curMax > globalMax) {
                 globalMax = curMax;
             }
         if(curMax < 0) {
               curMax = 0;
           }  

      }  
       return globalMax; 
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
       int maxSum = Integer.MIN_VALUE;
       int maxSumSoFar = 0;
       for(int num : nums){
            maxSumSoFar += num;
           maxSum = Math.max(maxSum,maxSumSoFar);
           if(maxSumSoFar < 0){
            maxSumSoFar = 0;
           }  
       }
        return maxSum;  
    }
}